package/src/versions/types.ts

```diff
@@ -0,0 +1,17 @@
+import type { SemVer } from "semver";
+import type { VersionSpec } from "./resolved-spec";
+import type { UnresolvedVersionSpec } from "./unresolved-spec";
+
+// TODO: should we allow passing raw strings as version-likes?
```

Comment:
Upside is simplicity, downside is ease of making mistakes.
It's just a matter of adding `| string` if this is something we want.